app/assets/javascripts/discourse/app/controllers/password-reset.js

```diff
@@ -22,23 +22,13 @@ export default Controller.extend(PasswordValidation, {
     "model.security_key_required"
   ),
   otherMethodAllowed: readOnly("model.multiple_second_factor_methods"),
-  @discourseComputed("model.security_key_required")
-  secondFactorMethod(security_key_required) {
-    return security_key_required
-      ? SECOND_FACTOR_METHODS.SECURITY_KEY
-      : SECOND_FACTOR_METHODS.TOTP;
-  },
   passwordRequired: true,
   errorMessage: null,
   successMessage: null,
   requiresApproval: false,
   redirected: false,
   maskPassword: true,
-
-  init() {
-    this._super(...arguments);
-    this.set("selectedSecondFactorMethod", this.secondFactorMethod);
-  },
+  selectedSecondFactorMethod: SECOND_FACTOR_METHODS.TOTP,
```

Comment:
This still doesn't sit right for me because it's not reflecting reality; if they don't have TOTP enabled then it will be security key by default. Instead, why don't we set this in the route for reset password? It is a bit ugly, but it at least reflects reality. Confirming it works locally with system specs and manual testing. 

```
diff --git a/app/assets/javascripts/discourse/app/controllers/password-reset.js b/app/assets/javascripts/discourse/app/controllers/password-reset.js
index 6ce6d5d3a0..1eccd2f177 100644
--- a/app/assets/javascripts/discourse/app/controllers/password-reset.js
+++ b/app/assets/javascripts/discourse/app/controllers/password-reset.js
@@ -6,7 +6,6 @@ import DiscourseURL, { userPath } from "discourse/lib/url";
 import { modKeysPressed } from "discourse/lib/utilities";
 import { getWebauthnCredential } from "discourse/lib/webauthn";
 import PasswordValidation from "discourse/mixins/password-validation";
-import { SECOND_FACTOR_METHODS } from "discourse/models/user";
 import getURL from "discourse-common/lib/get-url";
 import discourseComputed from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
@@ -28,7 +27,7 @@ export default Controller.extend(PasswordValidation, {
   requiresApproval: false,
   redirected: false,
   maskPassword: true,
-  selectedSecondFactorMethod: SECOND_FACTOR_METHODS.TOTP,
+  selectedSecondFactorMethod: alias("model.selectedSecondFactorMethod"),

   @discourseComputed()
   continueButtonText() {
@@ -68,9 +67,7 @@ export default Controller.extend(PasswordValidation, {
           password: this.accountPassword,
           second_factor_token:
             this.securityKeyCredential || this.secondFactorToken,
-          second_factor_method: this.securityKeyCredential
-            ? SECOND_FACTOR_METHODS.SECURITY_KEY
-            : this.selectedSecondFactorMethod,
+          second_factor_method: this.selectedSecondFactorMethod,
           timezone: moment.tz.guess(),
         },
       })
diff --git a/app/assets/javascripts/discourse/app/routes/password-reset.js b/app/assets/javascripts/discourse/app/routes/password-reset.js
index ee4cfb4faf..b15e184abd 100644
--- a/app/assets/javascripts/discourse/app/routes/password-reset.js
+++ b/app/assets/javascripts/discourse/app/routes/password-reset.js
@@ -1,6 +1,7 @@
 import { ajax } from "discourse/lib/ajax";
 import PreloadStore from "discourse/lib/preload-store";
 import { userPath } from "discourse/lib/url";
+import { SECOND_FACTOR_METHODS } from "discourse/models/user";
 import DiscourseRoute from "discourse/routes/discourse";
 import { deepMerge } from "discourse-common/lib/object";
 import I18n from "discourse-i18n";
@@ -21,6 +22,10 @@ export default class PasswordReset extends DiscourseRoute {
   afterModel(model) {
     // confirm token here so email clients who crawl URLs don't invalidate the link
     if (model) {
+      model.selectedSecondFactorMethod = model.security_key_required
+        ? SECOND_FACTOR_METHODS.SECURITY_KEY
+        : SECOND_FACTOR_METHODS.TOTP;
+
       return ajax({
         url: userPath(`confirm-email-token/${model.token}.json`),
         dataType: "json",
```